### PR TITLE
Feature scoreboard model

### PR DESCRIPTION
**Pull request by dlbucci**
THIS IS NOT READY, DO NOT MERGE.

Just sending this to get a feel for what people think.  This will require changes to config files, but 213 is the only course that uses any of these functions.

### CODE REVIEW

File: app/models/scoreboard.rb [hunk 1]
```diff
@@ -1,7 +1,10 @@
-class ScoreboardSetup < ActiveRecord::Base
```
Comment (icanberk) [inline app/models/scoreboard.rb]: I'd have preferred `ScoreboardSetup`. When you say Scoreboard I expect something that saves the scores of the users etc.

Comment (dlbucci) [inline app/models/scoreboard.rb]: This is more in line with the Autograder, and it makes more sense from a modeling perspective.  An Assessment has a Scoreboard or doesn't, and if it has a Scoreboard, then that scoreboard's settings will be in the Scoreboard model.  Plus, this is shorter, which is a plus in my book.

Comment (droh) [inline app/models/scoreboard.rb]: We'll adapt 213 config files to whatever you think best. So don't let 213
stand in the way of improving the scoreboard code.

On Saturday, April 11, 2015, Daniel Bucci <notifications@github.com> wrote:

> In app/models/scoreboard.rb
> <https://github.com/autolab/Autolab/pull/454#discussion_r28196609>:
>
> > @@ -1,7 +1,10 @@
> > -class ScoreboardSetup < ActiveRecord::Base
>
> This is more in line with the Autograder, and it makes more sense from a
> modeling perspective. An Assessment has a Scoreboard or doesn't, and if it
> has a Scoreboard, then that scoreboard's settings will be in the Scoreboard
> model. Plus, this is shorter, which is a plus in my book.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autolab/Autolab/pull/454/files#r28196609>.
>


-- 
Typed with fat fingers from my iPhone.

Comment (dlbucci) [inline app/models/scoreboard.rb]: It won't.  I've already change the scoreboardHeader spec, so we'll need to update those calls, but only the 213 assignments and one other assignment use that, so we'll be fine.

**Comment by dlbucci** [comment]
So I think this is about ready to merge.  People should pull and test though, this turned into a huge PR.

**Comment by jez** [comment]
So I see you fixed #289: does this mean we've fixed all that we needed to regarding refactoring config files, or did you just happen to be able to fix that issue?

**Comment by dlbucci** [comment]
It means we're gonna need to alter about 8 assessment config files, but besides those, we should be fine.

**Pull request merged**